06 MoveBackToStateless/src/nameEdit.jsx

```diff
@@ -1,19 +1,26 @@
 import React from 'react';
+import PropTypes from 'prop-types';
 
 export const NameEditComponent = (props) => {
-  return (
-    <div>
-      <label>Update Name:</label>
-      <input value={props.editingUserName}
-        onChange={(e) => props.onEditingNameUpdated(e.target.value)} />
-      <input type="submit" value="Change" className="btn btn-default"
-        onClick={props.onNameUpdateRequest} />
-    </div>
-  );
-}
+      return (
+        <div>
+          <label>Update Name:</label>
+          <input 
+            value={props.editingUserName}
+            onChange={(e) => props.onEditingNameUpdated(e.target.value)} 
```

Comment:
You can enhance this sample extracting the function from the tag, to avoid recreating the function on each render. what do you think?

Reply:
Tips, 

Step 1: first place it inside NameEditComponent

```diff
export const NameEditComponent = (props) => {
+      const = onChangeName(e) => {
+          props.onEditingNameUpdated(e.target.value)
+      }

      return (
        <div>
          <label>Update Name:</label>
          <input 
            value={props.editingUserName}
-            onChange={(e) => props.onEditingNameUpdated(e.target.value)} 
+             onChange={onChangeName} 
          />
          <input 
            type="submit"
            value="Change" 
            className="btn btn-default" 
            onClick={props.onNameUpdateRequest} 
          />
        </div>
      );
  }
```

By doing this our jsx markup is cleaner but still we are recreating the onChange function on each render let's go for approach 2 (next comment)

Reply:
We could extract the onChangeName function outside the component, what's the problem where, if we extract it we need the "props", that's why we have to use curry.

```diff
+      const = onChangeName(props) => (e) => {
+          props.onEditingNameUpdated(e.target.value)
+      }


export const NameEditComponent = (props) => {
-      const = onChangeName(e) => {
-          props.onEditingNameUpdated(e.target.value)
-      }

      return (
        <div>
          <label>Update Name:</label>
          <input 
            value={props.editingUserName}
-             onChange={onChangeName} 
+             onChange={onChangeName(props)} 
          />
          <input 
            type="submit"
            value="Change" 
            className="btn btn-default" 
            onClick={props.onNameUpdateRequest} 
          />
        </div>
      );
  }
```

Reply:
@HecFranco I think this is pending?